src/engine/strat_engine/backstore/devices.rs

```diff
@@ -237,9 +239,32 @@ pub fn initialize_devices(
         dev_info: &DeviceInfo,
         pool_uuid: PoolUuid,
         mda_data_size: MDADataSize,
+        key_description: Option<&str>,
     ) -> StratisResult<StratBlockDev> {
-        let mut f = OpenOptions::new().write(true).open(&dev_info.devnode)?;
+        let dev_uuid = Uuid::new_v4();
+        let (path, deviceno) = match key_description {
+            Some(desc) => {
+                let path = initialize_encrypted_stratis_device(
+                    &dev_info.devnode,
+                    pool_uuid,
+                    dev_uuid,
+                    desc,
+                )?;
+                let dm = DM::new()?;
+                let info = dm.device_info(&DevId::Name(DmName::new(&name_from_uuids(
+                    &pool_uuid, &dev_uuid,
+                ))?))?;
+                (path, info.device())
+            }
+            None => (dev_info.devnode.clone(), dev_info.devno),
+        };
+
+        let mut f = OpenOptions::new().write(true).open(&path)?;
 
+        // NOTE: May need to rethink hw_id in the context of an encrypted device -
```

Comment:
Let's put the ```hw_id``` question on the standup agenda.

Reply:
Sounds good.

Reply:
We decided in standup that encrypted devices never have a hw_id.

Reply:
Resolved.